### PR TITLE
Make dropout pass-through in !train mode.

### PR DESCRIPTION
**Pull request by dlibenzi**

### CODE REVIEW

**Comment by dlibenzi** [comment]
> Hmm this seems exactly like what I tried but didn't see it work properly. Did we test eval result is different with this if model.eval() is turned on?

I think we need to fix the eval path in our tensor sides first, then wire in eval() calls.
